spec/lib/archive_importer_spec.rb

```diff
@@ -138,14 +139,24 @@
           "show_community_spotlight_in_stream" => false,
           "language"                           => "ru",
           "disable_mail"                       => false,
-          "auto_follow_back"                   => true
+          "auto_follow_back"                   => true,
+          "blocks"                             => ["bad_guy@unknown.pod", person_to_block.diaspora_handle]
         }
       }
     }
 
     it "creates user" do
+      # an unknown blocked person
+      stub_request(:get, "https://unknown.pod/.well-known/webfinger?resource=acct:bad_guy@unknown.pod")
+        .to_return(status: 404, body: "", headers: {})
+      stub_request(:get, "https://unknown.pod/.well-known/host-meta")
+        .to_return(status: 404, body: "", headers: {})
+      stub_request(:get, "http://unknown.pod/.well-known/host-meta")
+        .to_return(status: 404)
+
       expect {
         archive_importer.create_user(username: "new_name", password: "123456")
+        archive_importer.import
```

Comment:
I am not sure.. shouldn't a 'create_user' create everything what is directly attached to a user, so the blocks should also be imported here? Thats why I added import_blocks call in the create method first. The user import is correctly tested if also blocks are imported/created, or am I wrong?
If I create a new blocks_importer_spec, I have the feeling that there I would also test it during the create_user call?

Reply:
No, otherwise you would need to test the whole import here. Contacts also aren't tested here and blocks are very similar to contacts (anti contacts?). The profile is a bit of a special case, since it's very closely attached to the user and is automatically created when you create the user anyway, so you as well can directly provide it with the correct data instead of first creating an empty profile and then insert the profile data later.